### ISSUE TITLE
krylov space re-use [BifurcationKit jacobian-free methods optimization]

### ISSUE THREAD

**Issue by rveltz** [Jutho/KrylovKit.jl#108]
Hi,

I am (kinda re-) opening an issue because I think it is different from the issues on memory allocations that are still open. In BifurcationKit.jl, we solve repeatedly linear systems and compute eigenvalues. **I am now improving BifurcationKit for Jacobian-free methods** by avoiding allocations when possible. I love KrylovKit because it is robust and have the most general matrix-free eigen solver.

Thus, it would be nice to allow

- for inplace linear operator
- re-use of Krylov subspace (memory) ; this could be increased if needed by `linsolve` or `eigsolve`. Is it something you have in mind?

**Comment by Jutho** [comment]
Yes I agree that this is becoming more pressing. I was kind of hoping that work on Julia's GC would make this unnecessary, but it seems like this is not going to happen any time soon.
